pre-miss/main.js

```diff
@@ -0,0 +1,84 @@
+import productService from "./ProductApi.js";
+import articleService from "./ArticleAPI.js";
+ 
+class Product {
+  constructor(name, description, price, tags, images, favoriteCount = 0) {
+    this.name = name;
+    this.description = description;
+    this.price = price;
+    this.tags = tags;
+    this.images = images;
+    this._favoriteCount = favoriteCount;
+  }

```

Comment:
```suggestion
  constructor(name, description, price, tags, images) {
    this.name = name;
    this.description = description;
    this.price = price;
    this.tags = tags;
    this.images = images;
    this._favoriteCount = 0
  }
```
좋아요 수에 언더바를 붙여서 변수를 만든것으로 보야 class 내부에서만 사용하기를 원하는 것으로 보여집니다.
그렇다면 생성자를 통해 인스턴스를 만들 때 favoritCount를 주입하는 것보다 생성자 함수 내부에서 초기화 해주는 것은 어떨까요?
이렇게 되면 생성 할 때뿐만 아니라, set을 통해서도 해당 변수에 값을 변화 시키지 못할 것으로 보여집니다.
단, 이러한 선택은 요구 조건에 따라 얼마든지 달라질 수 있습니다.